velox/common/memory/tests/HashStringAllocatorTest.cpp

```diff
@@ -678,5 +678,31 @@ TEST_F(HashStringAllocatorTest, clear) {
   EXPECT_EQ(allocator_->retainedSize(), 0);
 }
 
+TEST_F(HashStringAllocatorTest, freezeAndExecute) {
+  std::string str = "abc";
+  StringView view(str.data(), str.size());
+  allocator_->copyMultipart(view, reinterpret_cast<char*>(&view), 0);
+
+  str.clear();
+
+  // Freeing memory requires the HashStringAllocator to be mutable.
+  VELOX_ASSERT_THROW(
+      allocator_->freezeAndExecute([&]() {
+        allocator_->free(HashStringAllocator::headerOf(view.data()));
```

Comment:
Can you add a test that const method access is ok? Thanks!

Reply:
That was the intention of calling cumulativeBytes() at the bottom of this test.